src/test/groovy/info/solidsoft/gradle/pitest/PitestTaskConfigurationSpec.groovy

```diff
@@ -244,7 +244,11 @@ class PitestTaskConfigurationSpec extends BasicProjectBuilderSpec implements Wit
             project.sourceSets { intTest }
             project.pitest.testSourceSets = [project.sourceSets.intTest]
         expect:
-            task.taskArgumentMap()['classPath'] == assembleSourceSetsClasspathByNameAsStringSet("intTest").join(",")
+            task.taskArgumentMap()['classPath'] ==
+                (
+                    assembleSourceSetsClasspathByNameAsStringSet("intTest") +
+                    [new File(project.buildDir, "classes/java/main").absolutePath]
```

Comment:
I wonder that it started to be needed here. `classpath` didn't contain have `classes/java/main` before, assuming `java` plugin was applied? Or it was a limitation of this "unit" test (and in real execution it was included)?

Reply:
`intTest` source set is created from scratch and does not have any relation to `main` source set, so `intTest` classpath does not include any classes from `main`. I suppose previously such a project won't work with real Pitest, because Pitest would fail to find classes to mutate.

Reply:
I know what confused me.
```
 extension.mainSourceSets.set([javaSourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME)])
```
and:
```
    private Set<File> calculateBaseMutableCodePaths() {
        Set<SourceSet> sourceSets = extension.mainSourceSets.get()
        return sourceSets*.output.classesDirs.files.flatten() as Set<File>
    }
```
which you call should cover it. However, in the past it was used only for `mutableCodePaths`, not for `(additional)classpath`, what you fixed making it java-test-fixtures plugin compatible :).

Reply:
Btw, could you, making other changes, refactor `assembleSourceSetsClasspathByNameAsStringSet` to generate class only or class+resources classpath to keep `new File(project.buildDir, "classes/java/XXX")` in one place?

Reply:
I'm not sure what you wanted, but I changed the path handling.

Please check and close other discussions, I think I addressed all your comments.

Reply:
Sort of :). I will tune it up after merge.